compio-net/src/udp.rs

```diff
@@ -316,8 +316,21 @@ impl UdpSocket {
         .await
     }
 
+    /// Gets a socket option.
+    ///
+    /// # Safety
+    ///
+    /// The caller must ensure `T` is the correct type for `level` and `name`.
+    pub unsafe fn get_socket_option<T>(&self, level: i32, name: i32) -> io::Result<T> {
+        self.inner.get_socket_option(level, name)
+    }
+
     /// Sets a socket option.
-    pub fn set_socket_option<T>(&self, level: i32, name: i32, value: &T) -> io::Result<()> {
+    ///
+    /// # Safety
+    ///
+    /// The caller must ensure `T` is the correct type for `level` and `name`.
+    pub unsafe fn set_socket_option<T>(&self, level: i32, name: i32, value: &T) -> io::Result<()> {
```

Comment:
Maybe add a `Copy` bound? For `get` too. 
```suggestion
    pub unsafe fn set_socket_option<T: Copy>(&self, level: i32, name: i32, value: &T) -> io::Result<()> {
```